ECommerce.API/Controllers/ProductsController.cs

```diff
@@ -1,48 +1,61 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
-using Microsoft.AspNetCore.Http;
+﻿using ECommerce.Core.Abstractions;
+using ECommerce.Entities;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
-using ECommerce.DataAccess;
-using ECommerce.Entities;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace ECommerce.API.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
     public class ProductsController : ControllerBase
     {
-        private readonly EShopContext _context;
+        private readonly IProductsService _productsService;
 
-        public ProductsController(EShopContext context)
+        public ProductsController(IProductsService service)
         {
-            _context = context;
+            _productsService = service;
         }
 
         // GET: api/Products
         [HttpGet]
         public async Task<ActionResult<IEnumerable<Product>>> GetProducts()
         {
-            return await _context.Products.ToListAsync();
+            try
+            {
+                IEnumerable<Product> result = await _productsService.GetAllProducts();
+                return Ok(result);
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
         }
 
         // GET: api/Products/5
         [HttpGet("{id}")]
         public async Task<ActionResult<Product>> GetProduct(int id)
         {
-            var product = await _context.Products.FindAsync(id);
+            try
+            {
+                Product product = await _productsService.GetProduct(id);
 
-            if (product == null)
+                if (product == null)
+                {
+                    return NotFound();
+                }
+
+                return product;
+            }
+            catch (Exception ex)
             {
-                return NotFound();
+                throw ex;
```

Comment:
This will throw exception details to the client side instead we can return status Code 500 with descriptive message 
for example:
`return StatusCode(StatusCodes.Status500InternalServerError, "Failed to get product");`

Reply:
Done